http-client/src/main/java/no/nav/familie/http/client/AbstractRestClient.kt

```diff
@@ -91,6 +91,7 @@ abstract class AbstractRestClient(val operations: RestOperations,
             throw e
         } catch (e: Exception) {
             responsFailure.increment()
+            secureLogger.info("Feil ved kall mot uri=$uri: ${e.message}")
```

Comment:
Blir ikke dette nesten alltid logget likevel av de som bruker `AbstractRestClient` blir nok mye dobbelt av denne?

Reply:
Tja, vi har problemer med å få noe fornuftig logging på denne i ba-sak. 

[no.nav.familie.ba.sak.common.Feil]  En feil har oppstått: Feil ved oppslag på person. Gav feil: Feil ved kall mot uri=http://pdl-api/graphql
--

Reply:
Virker som message bare blir slukt.

Reply:
Men åpen for andre tiltak @blommish 👍

Reply:
Hmm, er ikke dette message som du ønsker da? https://logs.adeo.no/app/discover#/doc/tjenestekall-*/tjenestekall-team-teamfamilie-test-000014?id=h0iFInkBWnLJlrkWYjyV

Reply:
Eller nei, message er ikke der

Reply:
Nettopp, den blir bare slukt fordi man lager ny message :(

Reply:
Hvis alle catcher disse selv så er det jo kanskje bedre å bare kaste feilen videre uansett? 🤷

Reply:
Er ikke det pga att dere spiser feilen da
Hvis man legger inn throwable 
```
open class Feil(message: String,
                open val frontendFeilmelding: String? = null,
                open val httpStatus: HttpStatus = HttpStatus.OK,
                open val throwable: Throwable? = null) : RuntimeException(message, throwable)
```

Samt logger feil i frontendErrorResponse
```
secureLogger.error("$className En håndtert feil har oppstått(${feil.httpStatus}): " +
                           "${feil.frontendFeilmelding}, ${feil.stackTrace}", feil)
```

Så får man med hele stacktracen

Reply:
Ja, men i stacktracen så fremkommer ikke meldingen som blir kastet... Såvidt jeg vet? Vi logger hele stacktracen fra den meldingen jeg skrev i securelogger:

no.nav.familie.ba.sak.common.Feil: Feil ved oppslag på person. Gav feil: Feil ved kall mot uri=http://pdl-api/graphql
at no.nav.familie.ba.sak.pdl.PdlRestClient.hentPerson(PdlRestClient.kt:98)
, at no.nav.familie.ba.sak.pdl.PersonopplysningerService.hentPersoninfoMedQuery(PersonopplysningerService.kt:56)
, at no.nav.familie.ba.sak.pdl.PersonopplysningerService.hentPersoninfoMedRelasjoner(PersonopplysningerService.kt:25)
, at no.nav.familie.ba.sak.pdl.PersonopplysningerService$$FastClassBySpringCGLIB$$31ba5167.invoke(<generated>)
, at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)

Reply:
Eks
```
fun main() {
    val logger = LoggerFactory.getLogger("logger")
    try {
        a()
    } catch (feil: Feil) {
        val mostSpecificThrowable =
                if (feil.throwable != null) NestedExceptionUtils.getMostSpecificCause(feil.throwable!!) else null
        logger.error("asd, ${feil.stackTrace}")
        logger.error("----------------")
        logger.error("asd, ${feil.stackTrace}", mostSpecificThrowable)
        logger.error("----------------")
        logger.error("asd, ${feil.stackTrace}", feil)
    }
}

private fun a() {
    try {
        throw RuntimeException("olasd")
    } catch (e: Exception) {
        throw Feil("asd", "123", HttpStatus.MULTI_STATUS, e)
    }
}
```
Blir:
```
14:10:36.577 [main] ERROR logger - asd, [Ljava.lang.StackTraceElement;@166fa74d
14:10:36.581 [main] ERROR logger - ----------------
14:10:36.585 [main] ERROR logger - asd, [Ljava.lang.StackTraceElement;@4148db48 // dette er hvordan dere logger den idag
java.lang.RuntimeException: olasd
	at no.nav.familie.ba.sak.common.RessursUtilsKt.a(RessursUtils.kt:120)
	at no.nav.familie.ba.sak.common.RessursUtilsKt.main(RessursUtils.kt:106)
	at no.nav.familie.ba.sak.common.RessursUtilsKt.main(RessursUtils.kt)
14:10:36.585 [main] ERROR logger - ----------------
14:10:36.585 [main] ERROR logger - asd, [Ljava.lang.StackTraceElement;@3e77a1ed //hvordan man kan logge den
no.nav.familie.ba.sak.common.Feil: asd
	at no.nav.familie.ba.sak.common.RessursUtilsKt.a(RessursUtils.kt:122)
	at no.nav.familie.ba.sak.common.RessursUtilsKt.main(RessursUtils.kt:106)
	at no.nav.familie.ba.sak.common.RessursUtilsKt.main(RessursUtils.kt)
Caused by: java.lang.RuntimeException: olasd
	at no.nav.familie.ba.sak.common.RessursUtilsKt.a(RessursUtils.kt:120)
	... 2 common frames omitted
```

Reply:
Koden kommer hit, til else branchen:

```
catch (e: Exception) {
            when (e) {
                is Feil -> throw e
                else -> throw Feil(message = "Feil ved oppslag på person. Gav feil: ${e.message}",
                                   frontendFeilmelding = "Feil oppsto ved oppslag på person $personIdent",
                                   httpStatus = HttpStatus.INTERNAL_SERVER_ERROR,
                                   throwable = e)
            }
        }
```

Og `e.message` der er bare feilmeldingen som blir laget av AbstractRestClient :(

Reply:
Ja men `e` sin "cause" her er ju fortsatt `throw RuntimeException("Feil ved kall mot uri=$uri", e)` fra `AbstractRestClient`
Slik att når man logger Feil sin throwable, så burde den komme med hvis man ønsker.

Jeg er ikke helt sikker på att denne logger det dere ønsker
```
secureLogger.error("$className En håndtert feil har oppstått(${feil.httpStatus}): " +
                           "${feil.frontendFeilmelding}, ${feil.stackTrace}", throwable)
```

Selve endringen er ju ingen big issue, men for oss kommer denne endringen logge stacktrace 2 ganger :P

Reply:
Jeg såg ju faktiskt på feil metode... 
Det er ju `errorResponse` det handler om, og der er loggingen
`secureLogger.error("$className En feil har oppstått: $errorMessage", throwable)`

Så denne feilen blir ju håndtert i hentFagsakDeltager, slik jeg tolker det wrapper Feil i 
`throw IllegalStateException("Feil ved henting av person fra PDL", it)`

Denne håndteres sen i 
```fun handleThrowable(throwable: Throwable): ResponseEntity<Ressurs<Nothing>> {```

Der er ju mostSpecific Feil, og når man logger den så har man ikke feilmeldingen fra feilet fra AbstractRestClient.
Så tenker fortsatt det er mer riktig å legge inn stacktrace i super på Feil, dvs `: RuntimeException(message, throwable)`

Jeg er ikke 100% sikker på dette, men min magefølelse sier att det er mer riktig :P